### PR TITLE
Adds new Advanced Behavioral Detections section

### PR DESCRIPTION
**Pull request by natasha-moore-elastic**
Contributes to #4227.

Summary of changes:

- The ['Entity Risk Scoring'](https://www.elastic.co/guide/en/security/current/advanced-entity-analytics-overview.html#entity-risk-scoring) section has been moved and is now a child page of 'Advanced Entity Analytics'.
- The ['Turn on the risk scoring engine'](https://www.elastic.co/guide/en/security/current/turn-on-risk-engine.html) and ['View and analyze risk score data'](https://www.elastic.co/guide/en/security/current/analyze-risk-score-data.html) pages have been moved and are now child pages of the new 'Entity Risk Scoring' page.
- A new 'Advanced Behavioral Detections' page has been added as a child page of 'Advanced Entity Analytics'.
- Existing ['Anomaly detection with machine learning'](https://www.elastic.co/guide/en/security/current/machine-learning.html) page has been renamed to 'Anomaly detection' and is now a child page of 'Advanced Behavioral Detections'.
- New 'Behavioral detection use cases' page has been added as a child page of 'Advanced Behavioral Detections'.
- The ['Jobs in Advanced Analytics (UEBA) Elastic integrations'](https://www.elastic.co/guide/en/security/current/machine-learning.html#ml-integrations) section has been renamed to 'Elastic Integrations for behavioral detection use cases' and moved to the new 'Behavioral detection use cases' page.
- ['Prebuilt job reference'](https://www.elastic.co/guide/en/security/current/prebuilt-ml-jobs.html) and ['Optimizing anomaly results'](https://www.elastic.co/guide/en/security/current/tuning-anomaly-results.html) are now child pages of 'Advanced Behavioral Detections' (since they couldn't be nested under 'Anomaly detection' due to page level limitations in the TOC).

Previews:
- [Advanced Entity Analytics](https://security-docs_4296.docs-preview.app.elstc.co/guide/en/security/master/advanced-entity-analytics-overview.html)
- [Entity Risk Scoring](https://security-docs_4296.docs-preview.app.elstc.co/guide/en/security/master/entity-risk-scoring.html)
- [Advanced Behavioral Detections](https://security-docs_4296.docs-preview.app.elstc.co/guide/en/security/master/advanced-behavioral-detections.html)
- [Behavioral detection use cases](https://security-docs_4296.docs-preview.app.elstc.co/guide/en/security/master/behavioral-detection-use-cases.html)

### CODE REVIEW

File: docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc:6]:
```suggestion
Advanced Behavioral Detections includes two key capabilities:
```

File: docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc:4]:
```suggestion
Elastic's {ml} capabilities and advanced correlation, scoring, and visualization techniques can help you identify potential behavioral threats that may be associated with security incidents.
```

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:4]:
```suggestion
Behavioral detection identifies potential internal and external threats based on user and host activity. It employs a threat-centric approach to flag deviations from the norm and suspicious activity by analyzing patterns, anomalies, and context enrichment. 
```
I'm wondering about the capitalization / proper name for this set of features, since on the Advanced Behavioral Detections page, you capitalized "Advanced Behavioral Detections", but that full name isn't used here. Not sure what's correct for this feature but we should be consistent. In other words I'm wondering if this paragraph should start with "Advanced Behavioral Detections identify..."?

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:6]: This reads like we're saying Elastic builds use cases, which doesn't sound right to me. We build a feature or capability that _has_ use cases. Also I think its redundant to say machine learning and AI, feels buzzword-y. 
```suggestion
{elastic-sec}'s behavioral detection feature builds on its foundational SIEM detection capabilities, leveraging {ml} algorithms to enable proactive threat detection and hunting.
```

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:12]: Reading this from the perspective of a naive user, I'm not clear on what "packages" means. (A software package, or a set of capabilities targeted at a particular use case? Both?). Reading the links below helps clarify, but I think this page could keep it simple and just refer to capabilities. Users could then learn about each package when they click through to a particular integration.

Also, do the integrations "streamline" the deployment of ML packages (capabilities?) or do they "enable" them? (Is there a less streamlined way to enable them that doesn't use the integrations? I'm not sure).

I also suggest taking out the link on "integrations" because we already link to several specific integrations below, and I'm not sure we want to direct the user to the general integrations page from here.
```suggestion
Behavioral detection integrations provide a convenient way to enable behavioral detection capabilities. They streamline the deployment of components that implement behavioral detection, such as data ingestion, transforms, rules, {ml} jobs, and scripts.
```

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:21]:
```suggestion
The following is a list of behavioral detection integrations for various use cases:
```

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:8]:
```suggestion
Entity Risk Scoring allows you to monitor changes in the risk score of hosts and users in your environment. The risk scoring engine assesses risk based on data from various sources.
```
The sentence with "factoring" wasn't quite holding together. I'm not sure what all sources it uses so I just suggested various.

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:10]:
```suggestion
It leverages the Elastic SIEM detection engine to generate host and user risk scores from the last 30 days.
```
Maybe I'm overzealous but the first sentence seemed pretty marketing to me. The remaining sentence could go at the beginning of the next paragraph.

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (benironside) [inline docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:12]:
```suggestion
It also generates risk scores on a recurring interval, and allows for easy onboarding and management. The engine is built to factor in risks from all {elastic-sec} use cases, and allows you to customize and control how and when risk is calculated.
```
"data from... use-cases" seems off to me. Maybe replace with "Data from various sources" or "Data from across SIEM"?

**Review by benironside — COMMENTED** [review]
I know this PR was mostly focused on reorg, but I ended up editing some of content that got moved. A few things jumped out at me, so I left some suggestions for your consideration. Feel free to take them with a grain of salt, I am certainly not the most informed about this set of features. Hope they're helpful, lmk if anything doesn't make sense.

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:4]: The full feature hierarchy is:
- Advanced Entity Analytics
  - Entity Risk Scoring
  - Advanced Behavioral Detections
    - Anomaly detection
    - Behavioral detection use cases
    
Essentially, anomaly detection and behavioral detection use cases are two capabilities of the Advanced Behavioral Detections feature. In a previous PR where Entity Risk Scoring was first introduced, it was agreed to capitalize it as a feature (similar to how we capitalize AI Assistant, for example). I capitalized Advanced Behavioral Detections since it's at the same hierarchy level as Entity Risk Scoring. I'm not sure if we classify anomaly detection and behavioral detection use cases as features – @SourinPaul, @jmikell821 what are your thoughts?

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:6]: I think you're right, but I'll defer to @SourinPaul to confirm.

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:12]: @SourinPaul is there any difference between a 'package' and 'integration'? If they are the same, then we may need to rephrase this section to avoid confusion.

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:8]: How does this sound @benironside, @SourinPaul?  
```suggestion
Entity Risk Scoring allows you to monitor changes in the risk score of hosts and users in your environment. When generating advanced scoring analytics, the risk scoring engine takes into account threats from its end-to-end XDR use cases, such as SIEM, cloud, and endpoint.
```

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:12]: I'm not sure which sentence this suggestion refers to?

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (jmikell821) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:4]: I am leaning towards sentence case for `Advanced behavioral detections` for consistency, unless @SourinPaul has a business need otherwise.

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (jmikell821) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:6]: Agree with this suggestion too ^.

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (jmikell821) [reply docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:8]: @natasha-moore-elastic this is good, just one tweak. 
```suggestion
Entity Risk Scoring allows you to monitor risk score changes of hosts and users in your environment. When generating advanced scoring analytics, the risk scoring engine utilizes threats from its end-to-end XDR use cases, such as SIEM, cloud, and endpoint.
```

**Review by jmikell821 — COMMENTED** [review]
Hi @natasha-moore-elastic - lots of good feedback here, but we could definitely use @SourinPaul's help to address some of the questions. Incorporate what you can, and once feedback is completely merged we'll do a final pass-through. Thanks for pulling this together!

File: docs/advanced-entity-analytics/entity-risk-scoring.asciidoc
Comment (SourinPaul) [reply docs/advanced-entity-analytics/entity-risk-scoring.asciidoc:8]: Reads great!

**Review by SourinPaul — APPROVED** [review]
Consider the comments. Looks great.

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc:4]: In that case, should we also lowercase `Entity risk scoring`?

**Comment by natasha-moore-elastic** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by jmikell821** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by natasha-moore-elastic** [comment]
It looks like the build is failing due to:
```
04:05:07 INFO:build_docs:Bad cross-document links:
04:05:07 INFO:build_docs:  Kibana [master]: packages/kbn-doc-links/src/get_doc_links.ts contains broken links to:
04:05:07 INFO:build_docs:   - en/security/master/advanced-entity-analytics-overview.html#entity-risk-scoring
```

@hop-dev, I see that in [this PR](https://github.com/elastic/kibana/pull/172172/files) we added a link to `advanced-entity-analytics-overview.html#entity-risk-scoring`. We're restructuring the documentation, and Entity risk scoring will be a separate page (`entity-risk-scoring.html`) once this PR is merged. Could we update the link in the `packages/kbn-doc-links/src/get_doc_links.ts` file to point to this new location?

File: docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/advanced-behavioral-detections.asciidoc]: To close the loop on this, per the slack convo, we're lowercasing both `advanced behavioral detections` and `entity risk scoring`.

File: docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc
Comment (natasha-moore-elastic) [reply docs/advanced-entity-analytics/behavioral-detection-use-cases.asciidoc]: To close the loop on this, per the slack convo, we're lowercasing both `advanced behavioral detections` and `entity risk scoring`.

**Comment by hop-dev** [comment]
@natasha-moore-elastic here is the PR https://github.com/elastic/kibana/pull/172560

**Comment by natasha-moore-elastic** [comment]
> @natasha-moore-elastic here is the PR [elastic/kibana#172560](https://github.com/elastic/kibana/pull/172560)

Thanks for the quick turnaround, @hop-dev, I appreciate it!

**Review by jmikell821 — APPROVED** [review]

**Pull request merged**
